src/system/behavior.cpp

```diff
@@ -107,7 +111,6 @@ void startup_sequence() {
 }
 
 void shutdown() {
-  isShutdown = true;
 
```

Comment:
For timing issues, the shutdown flag must be set as early as possible
```
const bool wasAlreadyShutdown = isShutdown;
isShutdown = true;
// some actions are to be done only if the system was power-on before
  if(!wasAlreadyShutdown) {
...
```

Reply:
```diff
diff --git a/src/system/behavior.cpp b/src/system/behavior.cpp
index 074ce86..426095d 100644
--- a/src/system/behavior.cpp
+++ b/src/system/behavior.cpp
@@ -112,6 +112,10 @@ void startup_sequence() {

 void shutdown() {

+  // flag system as powered down
+  const bool wasAlreadyShutdown = isShutdown;
+  isShutdown = true;
+
   // deactivate strip power
   pinMode(OUT_BRIGHTNESS, OUTPUT);
   ledpower::write_current(0);  // power down
(1/4) Stage this hunk [y,n,q,a,d,j,J,g,/,e,p,?]? y
@@ -128,7 +132,7 @@ void shutdown() {
   button::set_color(utils::ColorSpace::BLACK);

   // some actions are to be done only if the system was power-on before
-  if(!isShutdown) {
+  if(!wasAlreadyShutdown) {

     // save the current config to a file
     // (takes some time so call it when the lamp appear to be shutdown already)
(2/4) Stage this hunk [y,n,q,a,d,K,j,J,g,/,e,p,?]? y
@@ -138,9 +142,6 @@ void shutdown() {
     user::power_off_sequence();
   }

-  // flag system as powered down
-  isShutdown = true;
-
   // do not power down when charger is plugged in
   if (!charger::is_usb_powered()) {
     charger::shutdown();
```